kuka_rsi_hw_interface/src/xml_handler/xml_param.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Komáromi Sándor
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstring>
+#include <stdexcept>
+#include <iostream>
+
+#include "xml_handler/xml_param.hpp"
+
+namespace xml
+{
+int XMLParam::PrintParam(char * & buffer_it, int & size_left)
+{
+  int idx = 0;
+  switch (param_type_) {
+    case XMLType::BOOL:
+      idx = snprintf(buffer_it, size_left, "%u", GetParamValue<bool>());
```

Comment:
Same comment about locale here I believe.

Reply:
I believe you are talking about whether the `printf("%f")`  can cause problem with the commas and dots dilemma, here too.
I have just looked on cpp reference. According to that I think it always use dots.
https://en.cppreference.com/w/c/io/fprintf